lib/sync-srv.js

```diff
@@ -336,32 +350,41 @@ function processPending(dataset_id, dataset, params, cb) {
           syncUtil.doLog(dataset_id, 'verbose', ' READ for DELETE Success', params);
           syncUtil.doLog(dataset_id, 'silly', ' READ for DELETE Data : \n' + util.inspect(data), params);
 
-          var preHash = syncUtil.generateHash(pre);
-          var dataHash = syncUtil.generateHash(data);
+          //var preHash = syncUtil.generateHash(pre);
+          //var dataHash = syncUtil.generateHash(data);
 
-          syncUtil.doLog(dataset_id, 'verbose', 'DELETE Hash Check ' + uid + ' (client :: dataStore) = ' + preHash + ' :: ' + dataHash, params);
+          //console.log('delete action - sync : ' + preHash);
+          //console.log('delete action - sync : ' + dataHash);
 
-          if (!dataHash) {
-            //record has already been deleted
-            syncUtil.doLog(dataset_id, 'info', 'DELETE Already performed - uid=' + uid + ' : hash = ' + hash, params);
-            return addUpdate("applied", "delete", hash, uid, '', itemCallback);
-          }
-          else {
-            if (preHash === dataHash) {
-              DataSetModel.doDeleteHandler(dataset_id, uid, function (err) {
-                if (err) {
-                  syncUtil.doLog(dataset_id, 'warn', 'DELETE Failed - uid=' + uid + ' : err = ' + err, params);
-                  return addUpdate("failed", "delete", hash, uid, err, itemCallback);
-                }
-                syncUtil.doLog(dataset_id, 'info', 'DELETE Success - uid=' + uid + ' : hash = ' + hash, params);
+          syncUtil.asyncGenerateHash(pre, function(err, preHash) {
+            syncUtil.asyncGenerateHash(data, function(err, dataHash) {
+              console.log('delete action - async : ' + preHash);
+              console.log('delete action - async : ' + dataHash);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)